tests/posix/fs/src/test_fat_mount.c

```diff
@@ -18,17 +18,29 @@ static struct fs_mount_t fatfs_mnt = {
 	.fs_data = &fat_fs,
 };
 
-static int test_mount(void)
+void *test_mount(void)
 {
 	int res;
 
 	res = fs_mount(&fatfs_mnt);
 	if (res < 0) {
 		TC_PRINT("Error mounting fs [%d]\n", res);
-		return TC_FAIL;
+		zassert_equal(0, res, NULL);
```

Comment:
I haven't tried asserting in the setup function. I'm not sure what would happen. Can you try running `zassert_true(false, NULL)` here to make sure that this doesn't prevent other suites from running in the same test binary? You'll likely need to add another suite and a test to it.

Long term @aaronemassey is working on an API to improve this. You would call `zassume_equal` here which is basically saying "this is not what I'm testing, but I'm going to need to assume that this can work"

Reply:
I kind of prefer the gtest semantics where there is `ASSERT_EQ()` for "must be true otherwise skip remainder and fail" and `EXPECT_EQ()` for "test will fail but continue and see what else happens".

I guess it's basically the same though.

Reply:
I'm just going to use `__ASSERT_NO_MSG()` here until there is some resolution. Is there an open issue number I can reference in comments?

Reply:
Right, the plan is to have:

* Assume - a failed assumption will mark the test skipped. It is expected that another test will cover the assumed state.
* Assert - a failure will stop the test and math it failed (same as gtest)
* Expect - a failure will mark the test failed but continue execution

Assume cones from junit and I really like it to reduce noise. Your test needs X as a precondition to run, but another test actually tests that feature. If that feature breaks you don't want every test that uses it to fail. It makes it a triage nightmare.

I'll try to work on these APIs this week.